src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -667,74 +681,68 @@ protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
                 return columns.ToArray();
             }
 
-            private Delegate MakeGetter<T>(DataViewRow input, int iinfo)
+            private Delegate MakeGetter<TInput, TTransformed>(DataViewRow input, int iinfo)
             {
-                var getter = input.GetGetter<long>(input.Schema[_parent._column.Source]);
-                ValueGetter<T> result = (ref T dst) =>
+                var getter = input.GetGetter<TInput>(input.Schema[_parent._column.Source]);
+
+                ValueGetter<TTransformed> result = (ref TTransformed dst) =>
                 {
-                    long dateTime = default;
-                    getter(ref dateTime);
+                    TInput dateTimeColumn = default;
+                    getter(ref dateTimeColumn);
 
-                    if (!_cache.TryGetValue(dateTime, out TimePoint timePoint))
-                    {
-                        _cache[dateTime] = _parent._column.Transform(dateTime);
-                        _oldestKeys.Enqueue(dateTime);
-                        timePoint = _cache[dateTime];
-
-                        // If more than 100 cached items, remove 20
-                        if (_cache.Count > 100)
-                        {
-                            for (int i = 0; i < 20; i++)
-                            {
-                                long key;
-                                while (!_oldestKeys.TryDequeue(out key)) { }
-                                while (!_cache.TryRemove(key, out TimePoint removedValue)) { }
-                            }
-                        }
-                    }
+                    long dateTime;
+
+                    // Type already in Posix format
+                    if(typeof(TInput) == typeof(long))
```

Comment:
This type checking should happen outside of the getter delegate. There is no need to pay this price every time this method is called. See other transformer implementations for examples.

Reply:
Refactored my MakeGetter and moved the type checking out of the delegate.